notebooks/03_evaluate/evaluation_spark.ipynb

```diff
@@ -0,0 +1,136 @@
+        "metadata": {}
+    }, {
+        "cell_type": "code",
+        "source": ["df_true = pd.DataFrame(\n        {\n            COL_USER: [1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3],\n            COL_ITEM: [1, 2, 3, 1, 4, 5, 6, 7, 2, 5, 6, 8, 9, 10, 11, 12, 13, 14],\n            COL_RATING: [5, 4, 3, 5, 5, 3, 3, 1, 5, 5, 5, 4, 4, 3, 3, 3, 2, 1],\n        }\n    )\nratings = spark.createDataFrame(df_true)\n\nratings.filter(ratings[COL_USER] == 1).show()"],
+        "metadata": {
+            "trusted": true
+        },
+        "outputs": [],
+        "execution_count": 10
+    }, {
+        "cell_type": "markdown",
+        "source": ["Build the recommendation model using ALS on the training data.\n\nNote we set cold start strategy to 'drop' to ensure we don't get NaN evaluation metrics"],
+        "metadata": {}
+    }, {
+        "cell_type": "code",
+        "source": ["als = ALS(maxIter=5, regParam=0.01, userCol=COL_USER, itemCol=COL_ITEM, ratingCol=COL_RATING,\n          coldStartStrategy=\"drop\")\npredictions = als.fit(ratings).transform(ratings)\n\npredictions.show()"],
+        "metadata": {
+            "trusted": true
+        },
+        "outputs": [],
+        "execution_count": 12
+    }, {
+        "cell_type": "markdown",
+        "source": ["## 2 Evaluation metrics"],
+        "metadata": {}
+    }, {
+        "cell_type": "markdown",
+        "source": ["### 2.1 Rating metrics"],
+        "metadata": {}
+    }, {
+        "cell_type": "code",
+        "source": ["rmse = RegressionEvaluator(metricName=\"rmse\", labelCol=COL_RATING, predictionCol=\"prediction\").evaluate(predictions)\nprint(\"The RMSE is {}\".format(rmse))\n\nr2 = RegressionEvaluator(metricName=\"r2\", labelCol=COL_RATING, predictionCol=\"prediction\").evaluate(predictions)\nprint(\"The R2 is {}\".format(r2))\n\nmae = RegressionEvaluator(metricName=\"mae\", labelCol=COL_RATING, predictionCol=\"prediction\").evaluate(predictions)\nprint(\"The MAE is {}\".format(mae))\n\nmse = RegressionEvaluator(metricName=\"mse\", labelCol=COL_RATING, predictionCol=\"prediction\").evaluate(predictions)\nprint(\"The mse is {}\".format(mse))"],
+        "metadata": {},
+        "outputs": [],
+        "execution_count": 15
+    }, {
+        "cell_type": "markdown",
+        "source": ["### 2.2 Ranking metrics"],
+        "metadata": {}
+    }, {
+        "cell_type": "markdown",
+        "source": ["The following code fits the model, and creates the properly formated output dataset for ranking evaluation."],
+        "metadata": {}
+    }, {
+        "cell_type": "code",
+        "source": ["output = RankingAdapter(k=5, recommender=als).fit(ratings).transform(ratings)\n\noutput.show()"],
```

Comment:
I'm curious how these metrics are created and how they compare in terms of speed to the metrics we created. If they are similar, we should think of a way to make them common. 
FYI @yueguoguo

Reply:
@dciborow do these ranking metrics adapter rely on Scala implementation of mmlspark? We may want to discuss more about it. If they are functionally the same with the ones existing in the repo, we should demonstrate the necessity of having them.

Reply:
yes, all of the math is done in scala side of mmlspark. 
The goal of the notebook is to show Spark Devs who start off with the SparkMLLib guide how to use the advanced ranking metrics.